openmdao/core/system.py

```diff
@@ -4824,9 +4827,16 @@ def _get_var_meta(self, name, key):
         object
             The value stored under key in the metadata dictionary for the named variable.
         """
+        
         if self._problem_meta is not None:
-            meta_all = self._problem_meta['model_ref']()._var_allprocs_abs2meta
-            meta_loc = self._problem_meta['model_ref']()._var_abs2meta
+            if self._problem_meta['model_ref']() is not None:
+                model_ref = self._problem_meta['model_ref']()
+            else:
+                model_ref = None
```

Comment:
the `if/else` here is unnecessary?
this is equivalent to `model_ref = self._problem_meta['model_ref']()`?

Reply:
Whoops, yep, good point. Should be corrected now

Reply:
actually, I think I see what you intended to do here... did you mean this?
```
        if self._problem_meta is not None:
            model_ref = self._problem_meta['model_ref']()
        else:
            model_ref = None
```
since the following statement expects `model_ref` to be set even if `_problem_meta` is `None`

Reply:
Sorry, I think I jumped to changing things without thinking about what I was doing. Yes, what you've written above is correct and should have the behavior I intended. I've updated the latest commit accordingly.